cfpb/cfgov-refresh
# Updates for Google schema pilot

Google has asked for some updates to the schema pilot running on answer pages. This adds a feature flag that can be used with path conditions to wrap a page's full content in markup.

## Additions

- Adds a feature flag and conditionals in answer template to apply `HowTo` schema markup around entire content of an answer based on path
- Adds `h3` to whitelist

## Removals
- H1/H2 testing feature flag on answer page (this will be removed anyway in the open answer enhancements PR)


## Testing

1. Check out branch
2. Add a new condition for `HOW_TO_SCHEMA` flag: `path matches` `^/ask-cfpb/what-is-the-best-way-to-negotiate-a-settlement-with-a-debt-collector-en-1447/`
3. Navigate to [answer page 1447](http://localhost:8000/ask-cfpb/what-is-the-best-way-to-negotiate-a-settlement-with-a-debt-collector-en-1447/) and check its source. Verify that the answer content is wrapped in a div with attributes `itemscope="" itemtype="http://schema.org/HowTo"` and H1 has attribute `itemprop="name"`
4. Remove path condition and verify that the schema markup is no longer applied to containing div and H1 on answer page

## Screenshots
<img width="848" alt="Screen Shot 2019-05-06 at 9 48 05 PM" src="https://user-images.githubusercontent.com/778171/57272578-20febe00-7049-11e9-9805-20ef6a7b09c7.png">


<img width="481" alt="Screen Shot 2019-05-06 at 9 22 00 AM" src="https://user-images.githubusercontent.com/778171/57239320-68049900-6fe0-11e9-9602-b1a784bb4b17.png">
